identity_jose/src/jws/header.rs

```diff
@@ -77,20 +85,54 @@ impl JwsHeader {
     self.b64 = Some(value.into());
   }
 
+  /// Returns the additional parameters in the header.
+  pub fn custom(&self) -> Option<&BTreeMap<String, Value>> {
+    self.custom.as_ref()
+  }
+
+  /// Sets additional parameters in the header.
+  pub fn set_custom(&mut self, value: BTreeMap<String, Value>) {
+    self.custom = Some(value)
+  }
+
   /// Returns `true` if the header contains the given `claim`, `false` otherwise.
   pub fn has(&self, claim: &str) -> bool {
     match claim {
       "alg" => self.alg().is_some(),
       "b64" => self.b64().is_some(),
-      _ => self.common.has(claim),
+      _ => {
+        self.common.has(claim)
+          || self
+            .custom
+            .as_ref()
+            .map(|custom| custom.get(claim).is_some())
+            .unwrap_or(false)
+      }
     }
   }
 
   /// Returns `true` if none of the fields are set in both `self` and `other`.
   pub fn is_disjoint(&self, other: &JwsHeader) -> bool {
     let has_duplicate: bool = self.alg().is_some() && other.alg.is_some() || self.b64.is_some() && other.b64.is_some();
 
-    !has_duplicate && self.common.is_disjoint(other.common())
+    !has_duplicate && self.common.is_disjoint(other.common()) && self.is_custom_disjoint(other)
+  }
+
+  fn is_custom_disjoint(&self, other: &JwsHeader) -> bool {
+    if let Some(custom) = self.custom() {
+      for key in custom.keys() {
+        if let Some(custom) = other.custom() {
+          if custom.contains_key(key) {
+            return false;
+          }
+        } else {
+          return true;
+        }
+      }
+    } else {
+      return true;
+    }
+    true
```

Comment:
```suggestion
  /// Returns `true` if none of the fields are set in both `self.custom` and `other.custom`.
  fn is_custom_disjoint(&self, other: &JwsHeader) -> bool {
    match (&self.custom, &other.custom) {
      (Some(self_custom), Some(other_custom)) => {
        for self_key in self_custom.keys() {
          if other_custom.contains_key(self_key) {
            return false;
          }
        }
        true
      }
      _ => true,
    }
  }
```

This is a bit easier to read and we don't evaluate `if let Some(custom) = other.custom() {` in every loop iteration.